link_ids.py

```diff
@@ -62,14 +63,27 @@ def do_link_ids(c, remove=False):
             writer = csv.DictWriter(csvfile, fieldnames=header)
             writer.writeheader()
             final_records = []
-            for row in database.household_match_groups.find(no_cursor_timeout=True):
-                final_record = {}
-                for s in systems:
-                    record_id = row.get(s, None)
-                    if record_id is not None:
-                        final_record[s] = record_id[0]
-                        all_ids_for_households[s].remove(record_id[0])
-                final_records.append(final_record)
+            with client.start_session(causal_consistency=True) as session:
+                with database.household_match_groups.find(
+                    session=session, no_cursor_timeout=True
+                ) as cursor:
```

Comment:
Hmm I thought I had seen guidance on doing it this way but maybe not. I probably won't be able to find the exact source I was looking at again but like you said I don't think it will change anything to do it this way